CMakeLists.txt

```diff
@@ -2125,6 +2129,33 @@ if(ENABLE_H2O)
                 "$<$<NOT:$<BOOL:${ARCH_SUPPORTS_64BIT_ATOMICS}>>:H2O_NO_64BIT_ATOMICS>")
 endif()
 
+#
+# build lmdb
+#
+
+add_library(lmdb STATIC
+    src/vendored/lmdb/lmdb.h
+    src/vendored/lmdb/mdb.c
+    src/vendored/lmdb/midl.h
+    src/vendored/lmdb/midl.c
+)
+
+target_include_directories(lmdb PUBLIC
+    ${CMAKE_SOURCE_DIR}/src/vendored/lmdb
+)
+
+target_compile_options(lmdb PRIVATE
+    -Wno-all
+    -Wno-extra
+    -Wno-shadow
+    -Wformat
+)
+
+target_compile_definitions(lmdb PRIVATE
+    MDB_USE_POSIX_SEM
+    $<$<BOOL:${OS_LINUX}>:_GNU_SOURCE>
+)
```

Comment:
@ktsaou you might want to double check this (or add more defs) in case it causes any  issues on Windows.